### PR TITLE
Access table associations using the dot syntax

### PR DESCRIPTION
**Pull request by sdustinh**
Reference: #11366

### CODE REVIEW

File: tests/TestCase/ORM/TableTest.php [hunk 1]
```diff
@@ -511,6 +511,29 @@ function (Event $event, $query, $options) use ($expected) {
...
+    {
+        $groups = TableRegistry::get('Groups');
+        $members = TableRegistry::get('Members');
+        $groupsMembers = TableRegistry::get('GroupsMembers');
+
+        $groups->belongsToMany('Members');
+        $groups->hasMany('GroupsMembers');
+        $groupsMembers->belongsTo('Members');
+
+        $association = $groups->association('GroupsMembers.Members');
```
Comment (markstory) [inline tests/TestCase/ORM/TableTest.php]: Could we get a test for associations deeper than 2 levels.

**Comment by markstory** [comment]
Moved to 3.next as that is where new features that modify existing methods will land. You may need to rebase your changes @sdustinh

**Comment by sdustinh** [comment]
Derp. I wasn't even paying attention. That's my bad.

File: tests/TestCase/ORM/TableTest.php (hunk 1 above)
Comment (sdustinh) [reply tests/TestCase/ORM/TableTest.php]: Definitely. Do you want me to crawl the same associations I have set up there or mix it up?

**Pull request closed without merging**